src/cast_core/src/CrudActor.cpp

```diff
@@ -637,6 +637,9 @@ struct FindOperation : public BaseOperation {
         if (opNode["Options"]) {
             _options = opNode["Options"].to<mongocxx::options::find>();
         }
+        if(opNode["Projection"]){
+            _options.projection(opNode["Projection"].to<DocumentGenerator>(context,id)());
```

Comment:
Hmmm. The fact that this is implemented this way and is not done alongside the rest of the conversion from `Node` to `mongocxx::options::find` is a little bit surprising to me:

https://github.com/mongodb/genny/blob/9fab05fdaef3f58711e00344abdafa31a8fa2c85/src/cast_core/include/cast_core/actors/OptionsConversion.hpp#L125-L154

I am hoping we can move this to the OptionsConversion.hpp so it lives alongside the related code.

Another related topic. This currently invokes the `DocumentGenerator` once on construction. That's probably not what we want, assuming that we even want to use the `DocumentGenerator` here in the first place. You'll notice that for the filter, the document generator is invoked in `run()`. The consequence is that each time the find command is run against the server, it can do so with a different generated filter. I would argue that projection should have the same behavior.